example-app/src/main/java/com/adyen/checkout/example/ui/settings/composable/SettingsScreen.kt

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 3/9/2024.
+ */
+
+@file:OptIn(ExperimentalMaterial3Api::class)
+
+package com.adyen.checkout.example.ui.settings.composable
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.WindowInsets
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.ime
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.windowInsetsPadding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.adyen.checkout.example.R
+import com.adyen.checkout.example.ui.compose.UIText
+import com.adyen.checkout.example.ui.settings.model.EditSettingsData
+import com.adyen.checkout.example.ui.settings.model.SettingsCategory
+import com.adyen.checkout.example.ui.settings.model.SettingsIdentifier
+import com.adyen.checkout.example.ui.settings.model.SettingsItem
+import com.adyen.checkout.example.ui.settings.model.SettingsUIState
+import com.adyen.checkout.example.ui.settings.viewmodel.SettingsViewModel
+import com.adyen.checkout.example.ui.theme.ExampleTheme
+
+@Composable
+internal fun SettingsScreen(
+    onBackPressed: () -> Unit,
+    viewModel: SettingsViewModel = hiltViewModel(),
+) {
+    Scaffold(
+        modifier = Modifier.windowInsetsPadding(WindowInsets.ime),
```

Comment:
Is `windowInsetsPadding` necessary here or using `innerPadding` is already enough?

Reply:
I just followed what we do in other screens, but I don't think we need it for this screen 👍